src/rax.c

```diff
@@ -1379,30 +1395,23 @@ int raxIteratorNextStep(raxIterator *it, int noup) {
 
                 /* Try visiting the next child if there was at least one
                  * additional child. */
-                if (!it->node->iscompr && it->node->size > (old_noup ? 0 : 1)) {
-                    raxNode **cp = raxNodeFirstChildPtr(it->node);
-                    int i = 0;
-                    while (i < it->node->size) {
-                        debugf("SCAN NEXT %c\n", it->node->data[i]);
-                        if (it->node->data[i] > prevchild) break;
-                        i++;
-                        cp++;
-                    }
-                    if (i != it->node->size) {
-                        debugf("SCAN found a new node\n");
-                        raxIteratorAddChars(it,it->node->data+i,1);
-                        if (!raxStackPush(&it->stack,it->node)) return 0;
-                        memcpy(&it->node,cp,sizeof(it->node));
-                        /* Call the node callback if any, and replace the node
-                         * pointer if the callback returns true. */
-                        if (it->node_cb && it->node_cb(&it->node))
-                            memcpy(cp,&it->node,sizeof(it->node));
-                        if (it->node->iskey) {
-                            it->data = raxGetData(it->node);
-                            return 1;
-                        }
-                        break;
+                if (!it->node->iscompr && it->child_offset + (old_noup ? 0 : 1) != it->node->size) {
```

Comment:
I still think it->child_offset in range [0, node->size] is ambiguous.
For example, in the following case 
```
it->child_offset == 2
it->node->size == 3
old_noup == 0
```

At this point we will not continue to search the node pointed to by `child_offset`, although I haven't thought of a suitable test to verify it, but I think it could happen.

Reply:
I am going to take some time to synchronize my antirez/rax fork with our current PR. My fork includes a number of other enhancements useful for me, e.g. relative searches (like ethereum) and next/prev child ops. It will take me a few days but then I will have a better testing environment as well.